vault/plugin_catalog.go

```diff
@@ -235,18 +279,25 @@ func (c *PluginCatalog) getPluginClient(ctx context.Context, sys pluginutil.Runn
 		return nil, err
 	}
 
+	clientConn := rpcClient.(*plugin.GRPCClient).Conn
+
+	// Wrap rpcClient with our implementation so that we can inject the
+	// ID into the context
+	pc.clientConn = &pluginClientConn{
+		ClientConn: clientConn,
+		id:         id,
+	}
```

Comment:
Do we only want to wrap this if we are a multiplexed client?

Reply:
addressed in: https://github.com/hashicorp/vault/pull/13896/commits/3afa08873ad064657cadc39d9bce94e45f49392e